src/programs.c

```diff
@@ -637,5 +778,1001 @@ int uprobe__read_return_string(struct pt_regs *ctx)
+    ret = bpf_probe_read(&ptr, sizeof(u64), (void*) PTR + (ii * sizeof(u64)));  \
+    if (ret < 0)                                                                \
+    {                                                                           \
+        goto Next;                                                              \
+    }                                                                           \
+    else if ((void *) ptr == NULL)                                              \
+    {                                                                           \
+        goto Next;                                                              \
+    }                                                                           \
+    else                                                                        \
+    {                                                                           \
+        count = bpf_probe_read_str(&ev->u.v.value, VALUE_SIZE, (void *) ptr);   \
+        if (count == VALUE_SIZE) {                                              \
+            ev->u.v.truncated = TRUE;                                           \
+        }                                                                       \
+        if (count <= 0) {                                                       \
+            goto Next;                                                          \
+        }                                                                       \
+    }                                                                           \
+    push_telemetry_event(ev);                                       \
+    ii++;
+
+
+
+static __always_inline int enter_exec(syscall_pattern_type_t sp, int fd,
+                                      const char __user *filename,
+                                      const char __user *const __user *argv,
+                                      const char __user *const __user *envp,
+                                      int flags, struct pt_regs *ctx, u32 ppid,
+                                      u32 luid)
+{
+    u64 id = 0; // reuse ID to save space
+    ptelemetry_event_t pev = bpf_map_lookup_elem(&telemetry_stack, (u32 *) &id);
+    if (!pev) // this should never happen, but the verifier complains otherwise
+    {
+        goto Exit;
+    }
+    telemetry_event_t sev = {0};
+    __builtin_memcpy(&sev, pev, sizeof(telemetry_event_t));
+    ptelemetry_event_t ev = &sev;
+
+    id = bpf_get_prandom_u32();
+    ev->id = id;
+    ev->done = FALSE;
+    ev->telemetry_type = 0,
+    ev->u.v.value[0] = '\0';
+    ev->u.v.truncated = FALSE;
+
+    FILL_TELEMETRY_SYSCALL_EVENT(ev, sp);
+    u32 pid = pid_tgid >> 32;
+    ev->u.syscall_info.ppid = ppid;
+    ev->u.syscall_info.luid = luid;
+    push_telemetry_event(ev);
+
+    bpf_map_update_elem(&telemetry_ids, &pid, &id, BPF_ANY);
+
+
+    // explicit null check to satisfy the verifier
+    if (filename)
```

Comment:
prefer `if (!filename) goto Exit` less indentation

Reply:
Got it here and the same pattern in a few (all?) other places. 

750d697f2242e69233ef10fb19251be28b8c7bae